packages/netlify-cms-lib-widgets/src/stringTemplate.ts

```diff
@@ -190,13 +190,13 @@ export function compileStringTemplate(
         replacement = data.getIn(keyToPathArray(key), '') as string;
       }
 
+      const filterFunction = getFilterFunction(filter);
+      if (filterFunction) {
+        replacement = filterFunction(replacement);
+      }
+
       if (processor) {
-        return processor(replacement);
-      } else {
-        const filterFunction = getFilterFunction(filter);
-        if (filterFunction) {
-          replacement = filterFunction(replacement);
-        }
+        replacement = processor(replacement);
```

Comment:
Instead of the the early return if the `processor` arg is passed in, check for filters first and apply them.

Moved the `processor` call *after* filters since it looks like that is used to make the slugs safer for URL/filenames.